### DISCUSSION TITLE
Exchange current density options in COMSOL model

### DISCUSSION THREAD

**Discussion by cannotcomplain** [pybamm-team/PyBaMM#2238] [Q&A] [answered]
Hello PyBaMM Team, 

Let me say again how much I appreciate you creating and maintaining this software. 

Our group is hoping to move away from COMSOL simulations and more exclusively use PyBaMM but we are running into difficulties recreating your DFN COMSOL comparisons. 

In particular, we were curious how you handled the "Electrode Kinetics" in COMSOL? Until now, we have been trying to use the 'Lithium insertion' option for the "Kinetics expression type" which defines a leading coefficient `i_{0,ref}` to the Butler-Volmer equation (see first image below).

We believe that this is similar to how PyBaMM defines `j0` using the `j0_dimensional` function and normalizing it by `j0_ref_dimensional` in `pybamm/parameters/lithium_ion_parameters.py` (See second image below). But, we can't get the curves to match without scaling the COMSOL equations by approximately `L_x / a_k_typ` which seemed to be leftover in the non-dimensional scalings of the PyBaMM equations (found with Latexify). 

Any insights on how PyBaMM handles the exchange current density compared to COMSOL would be greatly appreciated. 

Thanks again and kind regards,
Jay Johnson

<img width="716" alt="image" src="https://user-images.githubusercontent.com/78364049/184947131-3264a04b-2947-43ea-a98e-d9ec07fa0c02.png">

<img width="655" alt="image" src="https://user-images.githubusercontent.com/78364049/184947801-c64e7cbd-439e-4b6d-b32c-5d3a63737983.png">

**Comment by brosaplanella** [comment, marked as answer]
Hi! You need to define your dimensional `j0` in your parameter set, which can be defined to match your COMSOL expression. For example, if you check `pybamm/input/parameters/lithium_ion/positive_electrodes/nmc_Chen2020/nmc_LGM50_electrolyte_exchange_current_density_Chen2020.py`, you can see how it is defined.

You shouldn't manipulate `lithium_ion_parameters.py` as that includes all the nondimensionalisations that PyBaMM uses which are strongly dependent from each other, so changing one can break many things.
